api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -333,17 +334,36 @@ public QueryJobConfiguration previewBigQueryJobConfig(DataSetPreviewRequest requ
 
   @Override
   public Map<String, QueryJobConfiguration> domainToBigQueryConfig(DataSetRequest dataSetRequest) {
+    DbDataset dbDataset;
+    if (CollectionUtils.isEmpty(dataSetRequest.getDomainValuePairs())) {
```

Comment:
Should we put a dataSetId on the dataSetRequest? This way if the dataSetId is null we use the DataSetRequest object otherwise read the dbDataSet from the db with the dataSetId.

Reply:
Done